src/twisted/logger/test/test_format.py

```diff
@@ -429,3 +430,94 @@ class Unformattable(object):
 
     def __repr__(self):
         return str(1 / 0)
+
+
+
+class CapturedError(Exception):
+    """
+    A captured error for use in format tests.
+    """
+
+
+
+class FormatEventWithTracebackTests(unittest.TestCase):
+    """
+    Tests for L{formatEventWithTraceback}, all of which ensure that the
+    returned type is a UTF-8.
+    """
+
+    def test_formatEventWithTraceback(self):
+        """
+        An event with a C{log_failure} key will have a traceback appended.
+        """
+        f = Failure(CapturedError("This is a fake error"))
+        event = {
+            "log_format": u"This is a test log message"
+        }
+        event["log_failure"] = f
+        eventText = formatEventWithTraceback(event)
+        self.assertIsInstance(eventText, unicode)
+        self.assertIn('Traceback', eventText)
+        self.assertIn('This is a fake error', eventText)
+        self.assertIn('This is a test log message', eventText)
+
+
+    def test_formatEmptyEventWithTraceback(self):
+        """
+        An event with an empty C{log_format} key appends a traceback from
+        the accompanying failure.
+        """
+        f = Failure(CapturedError("This is a fake error"))
+        event = {
+            "log_format": u""
+        }
+        event["log_failure"] = f
+        eventText = formatEventWithTraceback(event)
+        self.assertIsInstance(eventText, unicode)
+        self.assertIn('Traceback', eventText)
+        self.assertIn('This is a fake error', eventText)
+
+
+    def test_formatUnformattableWithTraceback(self):
+        """
+        An event with an unformattable value in the C{log_format} key still
+        has a traceback appended.
+        """
+        event = {
+            "log_format": "{evil()}",
+            "evil": lambda: 1 / 0,
+            Unformattable(): "gurk",
```

Comment:
This is extraneous to the stated purpose of the test, which is served by `{evil()}`.  Please remove it.

Reply:
@markrwilliams I agree, with concerns.  It tests a valid "unformattable" path, but also neglects one.  The first path would be caught by your suggested change, eliminating the `Unformattable()`.  The second path, exercised by the `Unformattable()` key occurs when the __repr__ call for the event throws.  My question -- do we only need to exercise the former, or should I split this into two tests, one for each?  For the sake of completeness, that seems like the right thing to do, thoughts?